disco/extensions/upgrade_simulation/upgrades/common_functions.py

```diff
@@ -1128,7 +1138,69 @@ def get_thermal_equipment_info(compute_loading, equipment_type, upper_limit=None
+        deciding_column_name = kwargs.get("deciding_column_name")
+        compute_loading = kwargs.get("compute_loading")
+        upper_limit = kwargs.get("upper_limit")
+        ignore_switch = kwargs.get("ignore_switch", False)
+    elif equipment_type == "voltage":  # pop out of dictionary since kwargs are also passed to voltage function
+        voltage_upper_limit = kwargs.pop("voltage_upper_limit")
+        voltage_lower_limit = kwargs.pop("voltage_lower_limit")
+        raise_exception = kwargs.pop("raise_exception")
+        deciding_column_dict = kwargs.pop("deciding_column_dict")
+    else:
+        raise Exception(f"Incorrect equipment type {equipment_type} provided. Possible values: line, transformer, voltage.")
+        
+    comparison_dict = {}
+    sol = dss.Solution  # opendss solution interface
+    dss.run_command("Set mode=yearly number=1")  # 1 power flow per solve and I guess stepsize remains the same
+    startH = 0
+    num_pts = len(dss.LoadShape.PMult())
+
+    time_step = 1 
+    for present_step in range(startH, num_pts, time_step):
+        key_name = "timepoint_"+str(present_step)
+        logger.debug(f"Present step: {present_step}")
+        dss.run_command("Set Controlmode=Static") 
+        dss.Text.Command(f"set hour = {present_step}")
+        sol.Solve()  # solves for specific hour that has been set
+        logger.debug(dss.LoadShape.Name())
+        logger.debug(dss.LoadShape.PMult()[present_step])
+        logger.debug(dss.Loads.Name())
+        logger.debug(sum(dss.CktElement.Powers()[::2]))
+        if (equipment_type.lower() == "transformer") or (equipment_type.lower() == "line"):
+            if equipment_type.lower() == "line":
+                loading_df = get_all_line_info_instance(compute_loading=compute_loading, upper_limit=upper_limit, ignore_switch=ignore_switch)
+            elif equipment_type.lower() == "transformer":
+                loading_df = get_all_transformer_info_instance(compute_loading=compute_loading, upper_limit=upper_limit)
+            loading_df.set_index("name", inplace=True)
+            comparison_dict[key_name] = loading_df
+        elif equipment_type.lower() == "voltage":
+            bus_voltages_df, undervoltage_bus_list, overvoltage_bus_list, buses_with_violations = get_bus_voltages_instance(
+                    voltage_upper_limit=voltage_upper_limit, voltage_lower_limit=voltage_lower_limit, raise_exception=raise_exception, **kwargs)
+            bus_voltages_df.set_index("name", inplace=True)
+            comparison_dict[key_name] = bus_voltages_df
     
+    # compare all dataframe, and create one that contains all worst loading conditions (across all multiplier conditions)
+    if (equipment_type.lower() == "transformer") or (equipment_type.lower() == "line"):
+        loading_df = compare_multiple_dataframes(comparison_dict, deciding_column_name, comparison_type="max")
+        return loading_df

```

Comment:
Can you make it so that this function always returns the same type? It's possible that you could make different functions for transformers/lines vs voltage. That would eliminate a lot of your conditional statements.